### ISSUE TITLE
Read the `build-system.requires` during pixi init in some way

### ISSUE THREAD

**Issue by olivier-lacroix** [prefix-dev/pixi#1109]
During pixi init from a pyproject.toml, it would be nice if pixi read the `build-system.requires` in some way, and maybe created a 'build' environment that would be used to build python packages

_Originally posted by @pavelzw in https://github.com/prefix-dev/pixi/issues/1084#issuecomment-2027067605_

**Comment by tdejager** [comment]
Sure, but then we also need to support no build isolation, so that uv actually uses that environment, and not solves its own: https://github.com/prefix-dev/pixi/issues/1124

**Comment by ruben-arts** [comment]
@tdejager Would the now be able to be introduced as well? Since we now have `no-build-isolation`.

**Comment by tdejager** [comment]
Yes, but we would need to either install them as pypi dependencies giving you a chicken and egg problem, as it might be part of the default prefix.

*or* we need to interpret them as PyPI dependencies which part of a bigger feature I suppose.
